pyrenew_hew/pyrenew_hew_model.py

```diff
@@ -307,292 +342,57 @@ def __init__(
         self.hosp_admit_obs_process_rv = hosp_admit_obs_process_rv
-        return None
+        self.wastewater_obs_process_rv = wastewater_obs_process_rv
 
-    def validate(self):  # numpydoc ignore=GL08
-        return None
+    def validate(self) -> None:  # numpydoc ignore=GL08
+        pass
 
     def sample(
         self,
-        n_observed_disease_ed_visits_datapoints=None,
-        n_observed_hospital_admissions_datapoints=None,
-        data_observed_disease_ed_visits=None,
-        data_observed_disease_hospital_admissions=None,
-        right_truncation_offset=None,
+        data: PyrenewHEWData = None,
+        sample_ed_visits: bool = False,
+        sample_hospital_admissions: bool = False,
+        sample_wastewater: bool = False,
     ) -> ArrayLike:  # numpydoc ignore=GL08
-        if (
-            n_observed_disease_ed_visits_datapoints is None
-            and data_observed_disease_ed_visits is None
-        ):
-            raise ValueError(
-                "Either n_observed_disease_ed_visits_datapoints or "
-                "data_observed_disease_ed_visits "
-                "must be passed."
-            )
-        elif (
-            n_observed_disease_ed_visits_datapoints is not None
-            and data_observed_disease_ed_visits is not None
-        ):
-            raise ValueError(
-                "Cannot pass both n_observed_disease_ed_visits_datapoints "
-                "and data_observed_disease_ed_visits."
-            )
-        elif n_observed_disease_ed_visits_datapoints is None:
-            n_observed_disease_ed_visits_datapoints = len(
-                data_observed_disease_ed_visits
-            )
-
-        if (
-            n_observed_hospital_admissions_datapoints is None
-            and data_observed_disease_hospital_admissions is not None
-        ):
-            n_observed_hospital_admissions_datapoints = len(
-                data_observed_disease_hospital_admissions
-            )
-        elif n_observed_hospital_admissions_datapoints is None:
-            n_observed_hospital_admissions_datapoints = 0
-
-        n_weeks_post_init = n_observed_disease_ed_visits_datapoints // 7 + 1
-        n_days_post_init = n_observed_disease_ed_visits_datapoints
-
         latent_infections = self.latent_infection_process_rv(
-            n_days_post_init=n_days_post_init,
-            n_weeks_post_init=n_weeks_post_init,
+            n_days_post_init=data.n_days_post_init,
         )
-
-        sample_ed_visits = True
-        sample_admissions = n_observed_hospital_admissions_datapoints > 0
-        sample_wastewater = False
+        n_init_days = self.latent_infection_process_rv.infection_initialization_process.infection_init_method.n_timepoints
+        first_latent_infection_dow = (
+            data.first_data_date_overall - datetime.timedelta(days=n_init_days)
+        ).weekday()
 
         sampled_ed_visits, sampled_admissions, sampled_wastewater = (
```

Comment:
Slight preference for
```suggestion
        sampled_ed_visits = sampled_admissions = sampled_wastewater = None
```
or
```suggestion
        sampled_ed_visits = None
        sampled_admissions = None
        sampled_wastewater = None
```